### DISCUSSION TITLE
How can I "rich colorize" output sent to stdout.write ?

### DISCUSSION THREAD

**Discussion by ReddyKilowatt** [Textualize/rich#1589] [Q&A]
The Python cmd module 
https://docs.python.org/3.6/library/cmd.html

has several methods which send output to stdout.write (which is aliased to sys.stdout) as in ....

self.stdout.write("some text")

I want to rich colorize a program that I'm writing but don't quite
understand how to use Console() with the cmd module, so that the self.stdout.write() calls will be colorized by Console()

**Comment by willmcgugan** [comment]
There is a FileProxy class in Rich which is a file-like object which prints to the Console. You could replace your stdout with an instance of that, but you might have to tweak it to permit highlighting / markup.

**Reply by peppedilillo** [reply]
Hello Will, any head up on how to tweak FileProxy to allow highlighting and markup? Thanks.
